migrations/eap8.0/eap7.0/src/main/java/org/jboss/migration/eap7/to/eap8/EAP7_0ToEAP8_0ServerMigrationProvider.java

```diff
@@ -0,0 +1,117 @@
+        final LegacySecurityConfigurationMigration<WildFlyServer10> legacySecurityConfigurationMigration = new LegacySecurityConfigurationMigration<>();
+        final ServerUpdate.Builders<WildFlyServer10> serverUpdateBuilders = new ServerUpdate.Builders<>();
+        return serverUpdateBuilders.serverUpdateBuilder()
+                .standaloneServer(serverUpdateBuilders.standaloneConfigurationBuilder()
+                        .subtask(new WildFly27_0MigrateJBossDomainProperties<>())
+                        .subtask(legacySecurityConfigurationMigration.getReadLegacySecurityConfiguration())
+                        .subtask(new RemoveUnsupportedExtensions<>())
+                        .subtask(new RemoveUnsupportedSubsystems<>())
+                        .subtask(new MigrateReferencedModules<>())
+                        .subtask(new WildFly26_0MigrateReferencedPaths<>())
+                        .subtask(new WildFly26_0MigrateVault<>())
+                        .subtask(legacySecurityConfigurationMigration.getRemoveLegacySecurityRealms())
+                        .subtask(new EAP7_0ToEAP8_0UpdateInfinispanSubsystem<>())
+                        .subtask(new EAP7_0ToEAP8_0UpdateUndertowSubsystem<>())
+                        .subtask(new EAP7_0ToEAP8_0UpdateJGroupsSubsystem<>())
+                        .subtask(legacySecurityConfigurationMigration.getEnsureBasicElytronSubsystem())
+                        .subtask(legacySecurityConfigurationMigration.getMigrateLegacySecurityRealmsToElytron())
+                        .subtask(legacySecurityConfigurationMigration.getMigrateLegacySecurityDomainsToElytron())
+                        .subtask(new AddCoreManagementSubsystem<>())
+                        .subtask(new AddDiscoverySubsystem<>())
+                        .subtask(new AddEESecuritySubsystem<>())
+                        .subtask(new WildFly26_0AddHealthSubsystem<>())
+                        .subtask(new EAP8_0AddMetricsSubsystem<>())
+                        .subtask(new AddSocketBindingMulticastAddressExpressions<>())
+                )
+                .domain(serverUpdateBuilders.domainBuilder()
+                        .domainConfigurations(serverUpdateBuilders.domainConfigurationBuilder()
+                                .subtask(new WildFly27_0MigrateJBossDomainProperties<>())
+                                .subtask(legacySecurityConfigurationMigration.getReadLegacySecurityConfiguration())
+                                .subtask(new RemoveUnsupportedExtensions<>())
+                                .subtask(new RemoveUnsupportedSubsystems<>())
+                                .subtask(new MigrateReferencedModules<>())
+                                .subtask(new WildFly26_0MigrateReferencedPaths<>())
+                                .subtask(legacySecurityConfigurationMigration.getRemoveLegacySecurityRealms())
+                                .subtask(new EAP8_0AddHostExcludes<>())
```

Comment:
This task may need some adjustments: IIRC we support just EAP 7.4 in domain of EAP 8.0, but there are also exclusions for EAP 6.2 - 7.3. Furthermore, for EAP 7.4 there is exclusion for `opentelemetry` subsystem which is not in the default configuration.

Do you want me to create a CMTOOL issue?

Reply:
Created https://issues.redhat.com/browse/CMTOOL-340

Reply:
Thanks for the JIRA, let's sync this task ops later later there, once we know the final host excludes config, which frequently has "last minute" changes